go/interpreter/lfvm/instructions.go

```diff
@@ -1175,264 +1203,20 @@ func opCall(c *context) {
-	if err != nil {
-		return
-	}
-	stack := c.stack
-	// Pop call parameters.
-	provided_gas, addr, value, inOffset, inSize, retOffset, retSize := stack.pop(), stack.pop(), stack.pop(), stack.pop(), stack.pop(), stack.pop(), stack.pop()
-
-	// Compute and charge gas price for call
-	arg_memory_size, err := neededMemorySize(c, inOffset, inSize)
-	if err != nil {
-		return
-	}
-	ret_memory_size, err := neededMemorySize(c, retOffset, retSize)
-	if err != nil {
-		return
-	}
-	needed_memory_size := arg_memory_size
-	if ret_memory_size > arg_memory_size {
-		needed_memory_size = ret_memory_size
-	}
-	base_gas := c.memory.ExpansionCosts(needed_memory_size)
-	if base_gas < 0 || c.gas < base_gas {
-		c.status = OUT_OF_GAS
-		return
-	}
-
-	// We need to check the existence of the target account before removing
-	// the gas price for the other cost factors to make sure that the read
-	// in the state DB is always happening. This is the current EVM behavior,
-	// and not doing it would be identified by the replay tool as an error.
-	toAddr := tosca.Address(addr.Bytes20())
-
-	// Charge for transferring value to a new address
-	if !value.IsZero() {
-		base_gas += tosca.Gas(params.CallValueTransferGas)
-	}
-
-	cost := callGas(c.gas, base_gas, provided_gas)
-
-	if warmAccess {
-		if !c.UseGas(base_gas + cost) {
-			return
-		}
-	} else {
-		// In case of a cold access, we temporarily add the cold charge back, and also
-		// add it to the returned gas. By adding it to the return, it will be charged
-		// outside of this function, as part of the dynamic gas, and that will make it
-		// also become correctly reported to tracers.
-		c.gas += coldCost
-		if !c.UseGas(base_gas + cost + coldCost) {
-			return
-		}
-	}
-
-	// first use static and dynamic gas cost and then resize the memory
-	// when out of gas is happening, then mem should not be resized
-	c.memory.EnsureCapacityWithoutGas(needed_memory_size)
-	if !value.IsZero() {
-		cost += tosca.Gas(params.CallStipend)
-	}
-
-	// Check that the caller has enough balance to transfer the requested value.
-	if !value.IsZero() {
-		balance := c.context.GetBalance(c.params.Recipient)
-		balanceU256 := new(uint256.Int).SetBytes32(balance[:])
-		if balanceU256.Lt(value) {
-			c.stack.pushEmpty().Clear()
-			c.return_data = nil
-			c.gas += cost // the gas send to the nested contract is returned
-			return
-		}
-	}
-
-	// Get the arguments from the memory.
-	args := c.memory.GetSlice(inOffset.Uint64(), inSize.Uint64())
-	ret, err := c.context.Call(tosca.CallCode, tosca.CallParameters{
-		Sender:      c.params.Recipient,
-		Recipient:   c.params.Recipient,
-		CodeAddress: toAddr,
-		Input:       args,
-		Value:       tosca.Value(value.Bytes32()),
-		Gas:         cost,
-	})
-
-	if err == nil {
-		if memSetErr := c.memory.Set(retOffset.Uint64(), retSize.Uint64(), ret.Output); memSetErr != nil {
-			c.SignalError(memSetErr)
-		}
-	}
+func opCall(c *context) {
+	genericCall(c, tosca.Call)
```

Comment:
nit: as mentioned before, static check here and then call with StaticCall if needed.